core/build.gradle

```diff
@@ -16,7 +16,10 @@ dependencies {
     compile project(':grpc-api'),
             libraries.gson,
             libraries.android_annotations,
-            libraries.perfmark
+            libraries.errorprone // to avoid using perfmark's dependency
+    compile (libraries.perfmark) {
+        exclude group: 'com.google.errorprone', module: 'error_prone_annotations'
+    }
```

Comment:
Gradle complained perfmark's errorprone annotation 2.3.3.

https://travis-ci.org/grpc/grpc-java/jobs/630091143?utm_medium=notification&utm_source=github_status

Reply:
Yes. You did the appropriate thing here, although we normally put a comment saying we are using version X instead of version Y (so that in the future we can tell if the exclude is old and should be removed).

Our build enforces dependency convergence because I've not found a better approach to detecting requireUpperBoundDeps violations from within Gradle (because they aren't a problem in Gradle...).

Reply:
Updated comment as "prefer our version to perfmark's 2.3.3".